src/pytest_plugins/execute/pre_alloc.py

```diff
@@ -544,31 +558,32 @@ def pre(
     # Yield the pre-alloc for usage during the test
     yield pre
 
-    # Refund all EOAs (regardless of whether the test passed or failed)
-    refund_txs = []
-    for idx, eoa in enumerate(pre._funded_eoa):
-        remaining_balance = eth_rpc.get_balance(eoa)
-        eoa.nonce = Number(eth_rpc.get_transaction_count(eoa))
-        refund_gas_limit = 21_000
-        tx_cost = refund_gas_limit * default_gas_price
-        if remaining_balance < tx_cost:
-            continue
-        refund_tx = Transaction(
-            sender=eoa,
-            to=sender_key,
-            gas_limit=21_000,
-            gas_price=default_gas_price,
-            value=remaining_balance - tx_cost,
-        ).with_signature_and_sender()
-        refund_tx.metadata = TransactionTestMetadata(
-            test_id=request.node.nodeid,
-            phase="cleanup",
-            action="refund_from_eoa",
-            target=eoa.label,
-            tx_index=idx,
-        )
-        refund_txs.append(refund_tx)
-    eth_rpc.send_wait_transactions(refund_txs)
+    if not skip_cleanup:
+        # Refund all EOAs (regardless of whether the test passed or failed)
```

Comment:
What is the context for this request? Are we sure that we want to skip cleanup even if the test failed or just when it passes? Without more context, this doesn't seem ideal.

Reply:
Indeed, thanks for bringing this up, I think we need to wait for Kamil to try this out to see if this is what he had in mind since I did not ask for details on this request.